rmw_iceoryx2_cxx doesn't build with gcc 13

### Problem

`rmw_iceoryx2_cxx` fails to build with GCC 13

It works with `clang 18`.

### Expected

`rmw_iceoryx2_cxx` builds with both GCC and Clang.

### Environment

- Ubuntu 24.04
- ROS 2 Jazzy

```bash
$ gcc --version
gcc (Ubuntu 13.2.0-23ubuntu4) 13.2.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ clang --version
Ubuntu clang version 18.1.3 (1ubuntu1)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

### Log

```bash
--- stderr: rmw_iceoryx2_cxx                              
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/node_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/subscriber_impl.hpp:18,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/iox2/subscriber_impl.cpp:10:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/publisher_impl.hpp:18,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/iox2/publisher_impl.cpp:10:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/guard_condition_impl.hpp:18,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/iox2/guard_condition_impl.cpp:10:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/guard_condition.cpp:17:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/publisher.cpp:20:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/node.cpp:20:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:90: CMakeFiles/rmw_iceoryx2_cxx.dir/src/iox2/guard_condition_impl.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/waitset_impl.hpp:19,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/waitset.cpp:17:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:160: CMakeFiles/rmw_iceoryx2_cxx.dir/src/iox2/subscriber_impl.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:146: CMakeFiles/rmw_iceoryx2_cxx.dir/src/iox2/publisher_impl.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:286: CMakeFiles/rmw_iceoryx2_cxx.dir/src/rmw/guard_condition.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:342: CMakeFiles/rmw_iceoryx2_cxx.dir/src/rmw/node.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:356: CMakeFiles/rmw_iceoryx2_cxx.dir/src/rmw/publisher.cpp.o] Error 1
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/node_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/iox2/node_impl.cpp:10:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/iox2/iceoryx2.cpp:10:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/subscription.cpp:21:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:426: CMakeFiles/rmw_iceoryx2_cxx.dir/src/rmw/waitset.cpp.o] Error 1
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/iox2/context_impl.cpp:10:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:132: CMakeFiles/rmw_iceoryx2_cxx.dir/src/iox2/node_impl.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:104: CMakeFiles/rmw_iceoryx2_cxx.dir/src/iox2/iceoryx2.cpp.o] Error 1
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/waitset_impl.hpp:19,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/iox2/waitset_impl.cpp:10:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/graph.cpp:26:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:76: CMakeFiles/rmw_iceoryx2_cxx.dir/src/iox2/context_impl.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:412: CMakeFiles/rmw_iceoryx2_cxx.dir/src/rmw/subscription.cpp.o] Error 1
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/graph.cpp: In function ‘rmw_ret_t rmw_get_publishers_info_by_topic(const rmw_node_t*, rcutils_allocator_t*, const char*, bool, rmw_topic_endpoint_info_array_t*)’:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/graph.cpp:355:14: warning: unused variable ‘result’ [-Wunused-variable]
  355 |     if (auto result = rmw_topic_endpoint_info_array_check_zero(publishers_info) != RMW_RET_OK) {
      |              ^~~~~~
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/graph.cpp: In function ‘rmw_ret_t rmw_get_subscriptions_info_by_topic(const rmw_node_t*, rcutils_allocator_t*, const char*, bool, rmw_topic_endpoint_info_array_t*)’:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/graph.cpp:446:14: warning: unused variable ‘result’ [-Wunused-variable]
  446 |     if (auto result = rmw_topic_endpoint_info_array_check_zero(subscriptions_info) != RMW_RET_OK) {
      |              ^~~~~~
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/context_impl.hpp:17,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/init.cpp:19:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
In file included from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/publisher_impl.hpp:18,
                 from /home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/src/rmw/gid.cpp:16:
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: error: type/value mismatch at argument 1 in template parameter list for ‘template<iox2::ServiceType S> class iox2::ServiceBuilder’
  146 | auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
      |                                                                                                      ^
/home/alexis/ros2/src/rmw_iceoryx2/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp:146:102: note:   expected a constant of type ‘iox2::ServiceType’, got ‘rmw::iox2::Iceoryx2::ServiceType’ {aka ‘iox2::ServiceType’}
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:314: CMakeFiles/rmw_iceoryx2_cxx.dir/src/rmw/init.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:272: CMakeFiles/rmw_iceoryx2_cxx.dir/src/rmw/graph.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:258: CMakeFiles/rmw_iceoryx2_cxx.dir/src/rmw/gid.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/rmw_iceoryx2_cxx.dir/build.make:174: CMakeFiles/rmw_iceoryx2_cxx.dir/src/iox2/waitset_impl.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:157: CMakeFiles/rmw_iceoryx2_cxx.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake: *** [Makefile:146: all] Error 2
---
Failed   <<< rmw_iceoryx2_cxx [4.75s, exited with code 2]

Summary: 5 packages finished [21.6s]
  1 package failed: rmw_iceoryx2_cxx
  3 packages had stderr output: iceoryx2_cxx rmw_iceoryx2_cxx rmw_iceoryx2_cxx_test_msgs
```

--- a/.github/actions/install-clang/action.yml
+++ b/.github/actions/install-clang/action.yml
@@ -1,8 +1,8 @@
-name: 'Install clang toolchain'
+name: 'Install clang'
 runs:
   using: "composite"
   steps:
-    - name: Install iceoryx dependencies and clang-tidy
+    - name: Install clang
       shell: bash
       run: |
         sudo wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -

--- a/.github/actions/install-gcc/action.yml
+++ b/.github/actions/install-gcc/action.yml
@@ -1,0 +1,12 @@
+name: 'Install GCC'
+runs:
+  using: "composite"
+  steps:
+    - name: Install GCC
+      shell: bash
+      run: |
+        apt-get update
+        apt-get install -y software-properties-common
+        add-apt-repository -y ppa:ubuntu-toolchain-r/test
+        apt-get update
+        apt-get install -y gcc-13 g++-13

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -71,95 +71,104 @@ jobs:
 #         git fetch origin main
 #         ./scripts/check-clang-tidy.sh warning-as-error diff-to-main
 
-  colcon-build-test-clang:
+  colcon-build-test:
     needs: [preflight-check, static-code-analysis]
+    strategy:
+      matrix:
+        compiler: [clang]
+        include:
+          - compiler: clang
+            cc: clang
+            cxx: clang++
     runs-on: ubuntu-latest
     container:
       image: ros:rolling
     steps:
-    - name: Install ros build dependencies
-      run: |
-        apt-get update
-        apt-get install -y \
-          python3-colcon-common-extensions \
-          python3-colcon-mixin \
-          python3-vcstool \
-          python3-rosdep \
-          libacl1-dev \
-          software-properties-common \
-          curl \
-          wget
+      - name: Install common build dependencies
+        run: |
+          apt-get update
+          apt-get install -y \
+            python3-colcon-common-extensions \
+            python3-colcon-mixin \
+            python3-vcstool \
+            python3-rosdep \
+            libacl1-dev \
+            software-properties-common \
+            curl \
+            wget
 
-    - name: Install rust toolchain
-      uses: dtolnay/rust-toolchain@stable
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@stable
 
-    - name: Create workspace
-      run: |
-        mkdir -p $WORKSPACE_DIR/src/rmw_iceoryx2
-        mkdir -p $WORKSPACE_DIR/src/test_interface_files
+      - name: Setup workspace
+        run: |
+          mkdir -p $WORKSPACE_DIR/src/rmw_iceoryx2
+          mkdir -p $WORKSPACE_DIR/src/test_interface_files
 
-    - name: Checkout test_interface_files
-      uses: actions/checkout@v4
-      with:
-        repository: ros2/test_interface_files
-        path: src/test_interface_files
-        ref: rolling
+      - name: Checkout test message interfaces
+        uses: actions/checkout@v4
+        with:
+          repository: ros2/test_interface_files
+          path: src/test_interface_files
+          ref: rolling
 
-    - name: Checkout current ref
-      uses: actions/checkout@v4
-      with:
-        path: src/rmw_iceoryx2
+      - name: Checkout current ref
+        uses: actions/checkout@v4
+        with:
+          path: src/rmw_iceoryx2
 
-    - name: Install clang toolchain
-      uses: ./src/rmw_iceoryx2/.github/actions/install-clang
+      - name: Install compiler toolchain
+        if: matrix.compiler == 'clang'
+        uses: ./src/rmw_iceoryx2/.github/actions/install-clang
 
-    - name: Import dependencies with VCS
-      run: |
-        cd $WORKSPACE_DIR
-        sed 's|git@github.com:|https://github.com/|g' src/rmw_iceoryx2/iceoryx.repos > _deps.repos
-        vcs import src < _deps.repos
+      - name: Checkout iceoryx
+        run: |
+          cd $WORKSPACE_DIR
 
-    - name: Build with colcon
-      run: |
-        cd $WORKSPACE_DIR
-        . /opt/ros/rolling/setup.sh
-        colcon build \
-          --cmake-args \
-            -DCMAKE_C_COMPILER=clang \
-            -DCMAKE_CXX_COMPILER=clang++ \
-            -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-            -DBUILD_TESTING=On \
-          --symlink-install \
-          --packages-select \
-            test_interface_files \
-            iceoryx_platform\
-            iceoryx_hoofs \
-            iceoryx2_cxx \
-            rmw_iceoryx2_cxx_test_msgs \
-            rmw_iceoryx2_cxx
+          # need to switch to https in ci
+          sed 's|git@github.com:|https://github.com/|g' src/rmw_iceoryx2/iceoryx.repos > _deps.repos
+          vcs import src < _deps.repos
 
-    - name: Run tests
-      run: |
-        cd $WORKSPACE_DIR
-        . /opt/ros/rolling/setup.sh
-        . install/setup.sh
-        colcon test \
-          --packages-select \
-            rmw_iceoryx2_cxx \
-          --event-handlers console_direct+ \
-          --return-code-on-test-failure
+      - name: Build workspace
+        run: |
+          cd $WORKSPACE_DIR
+          . /opt/ros/rolling/setup.sh
+          colcon build \
+            --cmake-args \
+              -DCMAKE_C_COMPILER=${{ matrix.cc }} \
+              -DCMAKE_CXX_COMPILER=${{ matrix.cxx }} \
+              -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+              -DBUILD_TESTING=On \
+            --symlink-install \
+            --packages-select \
+              test_interface_files \
+              iceoryx_platform \
+              iceoryx_hoofs \
+              iceoryx2_cxx \
+              rmw_iceoryx2_cxx_test_msgs \
+              rmw_iceoryx2_cxx
 
-    - name: Generate test reports
-      if: always()
-      run: |
-        cd $WORKSPACE_DIR
-        colcon test-result --verbose
+      - name: Run tests
+        run: |
+          cd $WORKSPACE_DIR
+          . /opt/ros/rolling/setup.sh
+          . install/setup.sh
+          colcon test \
+            --packages-select rmw_iceoryx2_cxx \
+            --event-handlers console_direct+ \
+            --return-code-on-test-failure
 
-    - name: Upload test results
-      if: always()
-      uses: actions/upload-artifact@v4
-      with:
-        name: test-results
-        path: |
-          ${{ env.WORKSPACE_DIR }}/build/*/test_results/*/*.xml
-          ${{ env.WORKSPACE_DIR }}/log/**/test*.log
+      - name: Generate test reports
+        if: always()
+        run: |
+          cd $WORKSPACE_DIR
+          colcon test-result --verbose
+
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-results-${{ matrix.compiler }}
+          path: |
+            ${{ env.WORKSPACE_DIR }}/build/*/test_results/*/*.xml
+            ${{ env.WORKSPACE_DIR }}/log/**/test*.log

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -18,7 +18,7 @@ jobs:
       source-code: ${{ steps.filter.outputs.source-code }}
       markdown: ${{ steps.filter.outputs.markdown }}
     steps:
-      - name: Checkout sources
+      - name: Check out sources
         uses: actions/checkout@v4
 
       - name: Check for changed file types
@@ -37,7 +37,7 @@ jobs:
     timeout-minutes: 10
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout sources
+      - name: Check out sources
         uses: actions/checkout@v4
 
       - name: Check format of all commit messages
@@ -52,7 +52,7 @@ jobs:
     timeout-minutes: 10
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout sources
+      - name: Check out sources
         uses: actions/checkout@v4
 
       - name: Install clang toolchain
@@ -109,29 +109,40 @@ jobs:
           mkdir -p $WORKSPACE_DIR/src/rmw_iceoryx2
           mkdir -p $WORKSPACE_DIR/src/test_interface_files
 
-      - name: Checkout test message interfaces
+      - name: Check out test message interfaces
         uses: actions/checkout@v4
         with:
           repository: ros2/test_interface_files
           path: src/test_interface_files
           ref: rolling
 
-      - name: Checkout current ref
+      - name: Check out current ref
         uses: actions/checkout@v4
         with:
           path: src/rmw_iceoryx2
 
       # this is also required for all builds as iceoryx2 depends on libclang
-      - name: Install compiler toolchain
+      - name: Install clang toolchain
         uses: ./src/rmw_iceoryx2/.github/actions/install-clang
 
-      - name: Checkout iceoryx
+      - name: Install gcc toolchain
+        if: matrix.compiler == 'gcc'
+        uses: ./src/rmw_iceoryx2/.github/actions/install-gcc
+
+      - name: Import iceoryx
         run: |
           cd $WORKSPACE_DIR
 
           # need to switch to https in ci
           sed 's|git@github.com:|https://github.com/|g' src/rmw_iceoryx2/iceoryx.repos > _deps.repos
           vcs import src < _deps.repos
+
+      - name: Check compiler versions
+        run: |
+          echo "C compiler version:"
+          ${{ matrix.cc }} --version
+          echo "C++ compiler version:"
+          ${{ matrix.cxx }} --version
 
       - name: Build workspace
         run: |
@@ -159,7 +170,6 @@ jobs:
           . install/setup.sh
           colcon test \
             --packages-select rmw_iceoryx2_cxx \
-            --event-handlers console_direct+ \
             --return-code-on-test-failure
 
       - name: Generate test reports

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -74,12 +74,16 @@ jobs:
   colcon-build-test:
     needs: [preflight-check, static-code-analysis]
     strategy:
+      fail-fast: false
       matrix:
-        compiler: [clang]
+        compiler: [clang, gcc]
         include:
           - compiler: clang
             cc: clang
             cxx: clang++
+          - compiler: gcc
+            cc: gcc
+            cxx: g++
     runs-on: ubuntu-latest
     container:
       image: ros:rolling
@@ -117,8 +121,8 @@ jobs:
         with:
           path: src/rmw_iceoryx2
 
+      # this is also required for all builds as iceoryx2 depends on libclang
       - name: Install compiler toolchain
-        if: matrix.compiler == 'clang'
         uses: ./src/rmw_iceoryx2/.github/actions/install-clang
 
       - name: Checkout iceoryx

--- a/doc/release-notes/unreleased.md
+++ b/doc/release-notes/unreleased.md
@@ -11,8 +11,6 @@
     conflicts when merging.
 -->
 
-* Example text [#1](https://github.com/ekxide/rmw_iceoryx2/issues/1)
-
 ### Bugfixes
 
 <!--
@@ -20,7 +18,7 @@
     conflicts when merging.
 -->
 
-* Example text [#1](https://github.com/ekxide/rmw_iceoryx2/issues/1)
+* Fix failing `gcc` build [#15](https://github.com/ekxide/rmw_iceoryx2/issues/15)
 
 ### Refactoring
 
@@ -29,8 +27,6 @@
     conflicts when merging.
 -->
 
-* Example text [#1](https://github.com/ekxide/rmw_iceoryx2/issues/1)
-
 ### Workflow
 
 <!--
@@ -38,8 +34,9 @@
     conflicts when merging.
 -->
 
-* Use VCS to manage dependencies [#13](https://github.com/ekxide/rmw_iceoryx2/issues/13)
-* Add CI workflow for building and testing [#12](https://github.com/ekxide/rmw_iceoryx2/issues/12)
+* Use `vcs` to manage dependencies [#13](https://github.com/ekxide/rmw_iceoryx2/issues/13)
+* Add CI for building and testing with `clang` [#12](https://github.com/ekxide/rmw_iceoryx2/issues/12)
+* Add CI for building and testing with `gcc` [#15](https://github.com/ekxide/rmw_iceoryx2/issues/15)
 
 ### New API features
 

--- a/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp
+++ b/rmw_iceoryx2_cxx/include/rmw_iceoryx2_cxx/iox2/iceoryx2.hpp
@@ -142,15 +142,15 @@ private:
 
 // ===================================================================================================================
 
-template <::iox2::ServiceType ServiceType>
-auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<ServiceType> {
+template <::iox2::ServiceType S>
+auto Iceoryx2::service_builder(const std::string& service_name) -> ::iox2::ServiceBuilder<S> {
     auto name = ::iox2::ServiceName::create(service_name.c_str());
     if (name.has_error()) {
         // TODO: propagate error
     }
-    if constexpr (ServiceType == ::iox2::ServiceType::Local) {
+    if constexpr (S == ::iox2::ServiceType::Local) {
         return local().service_builder(name.value());
-    } else if constexpr (ServiceType == ::iox2::ServiceType::Ipc) {
+    } else if constexpr (S == ::iox2::ServiceType::Ipc) {
         return ipc().service_builder(name.value());
     } else {
         static_assert(std::false_type::value, "Attempted to build a service of unknown type");

--- a/rmw_iceoryx2_cxx/src/rmw/graph.cpp
+++ b/rmw_iceoryx2_cxx/src/rmw/graph.cpp
@@ -352,7 +352,7 @@ rmw_ret_t rmw_get_publishers_info_by_topic(const rmw_node_t* node,
     if (!rcutils_allocator_is_valid(allocator)) {
         return RMW_RET_INVALID_ARGUMENT;
     }
-    if (auto result = rmw_topic_endpoint_info_array_check_zero(publishers_info) != RMW_RET_OK) {
+    if (rmw_topic_endpoint_info_array_check_zero(publishers_info) != RMW_RET_OK) {
         return RMW_RET_INVALID_ARGUMENT;
     }
 
@@ -443,7 +443,7 @@ rmw_ret_t rmw_get_subscriptions_info_by_topic(const rmw_node_t* node,
     if (!rcutils_allocator_is_valid(allocator)) {
         return RMW_RET_INVALID_ARGUMENT;
     }
-    if (auto result = rmw_topic_endpoint_info_array_check_zero(subscriptions_info) != RMW_RET_OK) {
+    if (rmw_topic_endpoint_info_array_check_zero(subscriptions_info) != RMW_RET_OK) {
         return RMW_RET_INVALID_ARGUMENT;
     }
 

--- a/rmw_iceoryx2_cxx/test/test_rmw_allocator.cpp
+++ b/rmw_iceoryx2_cxx/test/test_rmw_allocator.cpp
@@ -17,18 +17,6 @@ namespace
 
 using namespace rmw::iox2::testing;
 
-class DummyClass
-{
-public:
-    DummyClass()
-        : value(42) {
-    }
-    ~DummyClass() {
-        value = 0;
-    }
-    int value;
-};
-
 class AllocatorHelpersTest : public TestBase
 {
 protected:
@@ -113,21 +101,37 @@ TEST_F(AllocatorHelpersTest, deallocate_nullptr) {
     ASSERT_NO_THROW(rmw::iox2::deallocate(ptr));
 }
 
+class DummyClass
+{
+public:
+    DummyClass() {
+        ++constructor_calls;
+    }
+    ~DummyClass() {
+        ++destructor_calls;
+    }
+
+    static size_t constructor_calls;
+    static size_t destructor_calls;
+};
+size_t DummyClass::constructor_calls = 0;
+size_t DummyClass::destructor_calls = 0;
+
 TEST_F(AllocatorHelpersTest, construct_and_destruct) {
     auto allocation = rmw::iox2::allocate<DummyClass>();
     ASSERT_TRUE(allocation.has_value());
 
     auto ptr = allocation.value();
     auto constructed = rmw::iox2::construct(ptr);
-    ASSERT_TRUE(constructed.has_value());
+    ASSERT_FALSE(constructed.has_error());
+    ASSERT_EQ(DummyClass::constructor_calls, 1u);
+    ASSERT_EQ(DummyClass::destructor_calls, 0u);
 
-    auto obj = constructed.value();
-    ASSERT_EQ(obj->value, 42);
+    rmw::iox2::destruct<DummyClass>(ptr);
+    ASSERT_EQ(DummyClass::constructor_calls, 1u);
+    ASSERT_EQ(DummyClass::destructor_calls, 1u);
 
-    rmw::iox2::destruct<DummyClass>(obj);
-    ASSERT_EQ(obj->value, 0);
-
-    rmw::iox2::deallocate(obj);
+    rmw::iox2::deallocate(ptr);
 }
 
 TEST_F(AllocatorHelpersTest, destruct_nullptr) {

--- a/rmw_iceoryx2_cxx/test/test_rmw_guard_condition.cpp
+++ b/rmw_iceoryx2_cxx/test/test_rmw_guard_condition.cpp
@@ -59,7 +59,7 @@ protected:
         auto listener = service.listener_builder().create().expect("failed to create test listener");
 
         return listener;
-    };
+    }
 
 private:
     iox::optional<::rmw::iox2::Iceoryx2> m_iox2;

--- a/rmw_iceoryx2_cxx/test/test_rmw_waitset.cpp
+++ b/rmw_iceoryx2_cxx/test/test_rmw_waitset.cpp
@@ -56,14 +56,14 @@ public:
 
     ~WaitSetTestContext() {
         for (size_t i = 0; i < m_subscriptions.size(); i++) {
-            (void)rmw_destroy_publisher(m_rmw_node, m_publishers.at(i).publisher);
-            (void)rmw_destroy_subscription(m_rmw_node, m_subscriptions.at(i).subscriber);
+            EXPECT_RMW_OK(rmw_destroy_publisher(m_rmw_node, m_publishers.at(i).publisher));
+            EXPECT_RMW_OK(rmw_destroy_subscription(m_rmw_node, m_subscriptions.at(i).subscriber));
         }
         for (size_t i = 0; i < m_guard_conditions.size(); i++) {
-            (void)rmw_destroy_guard_condition(m_guard_conditions.at(i));
+            EXPECT_RMW_OK(rmw_destroy_guard_condition(m_guard_conditions.at(i)));
         }
         if (m_waitset) {
-            (void)rmw_destroy_wait_set(m_waitset);
+            EXPECT_RMW_OK(rmw_destroy_wait_set(m_waitset));
         }
     }
 

